src/guides/components/endpoint.md

```diff
@@ -0,0 +1,253 @@
+
+## Backend
+
+The backend refers to what is available server-side, what is "_under-the-hood_" or "_out-of-sight_". By incorporating a backend aspect to your plugin, you can extend upon the server-side functionality. This includes storing the data produced by your plugin, sending HTTP requests, or processing user-supplied data.
+
+## sdk.api.register()
+
+This method allows you to register functions that can be called from the frontend. Meaning you can link a frontend action, like a button click, to the execution of some code.
+
+### Registering an API Endpoint
+
+For example, the following could be the contents of your plugin's `/packages/backend/src/index.ts` file:
+
+First, the necessary type aliases are imported. `SDK` is the interface used to interact with Caido. `DefineAPI` is used to structure the API: definining what methods or endpoints are available, the parameters those methods accept and what types of values they return.
+
+``` ts
+import { SDK, DefineAPI } from "caido:plugin";
+```
+
+Next, the function is defined. The function takes three parameters: `sdk`, `a` and `b`. The `sdk` parameter is typed using the `SDK` alias to give the function access to its utilities. The `a` and `b` parameters are expected to be numbers as this function multiplies the two together.
+
+``` ts
+function multiply(sdk: SDK, a: number, b: number) {
+    const result = a * b;
+    sdk.console.log(`The product of the multiply call is: ${result}`);
+    return result;
+}
+```
+
+Using the `DefineAPI` utility, we are stating what our API offers. In this case, the `multiply` function is available to be called. To ensure the function receives the expected parameter data types, `typeof` is used to link the `multiply` API call to the `multiply` function definition. This definition is stored in the type alias `API` and exported so it can be used in other files.
+
+``` ts
+export type API = DefineAPI<{
+    multiply: typeof multiply;
+}>;
+```
+
+Next, we define a function that will run as soon as Caido loads the plugin. It extends upon the base `SDK` by adding the `<API>`. In order to register the function, we use the `sdk.api.register()` method which takes two parameters: a string name for the function and the function it refers to. We give the name `"multiply"` to the `multiply` function.
+
+``` ts
+export function init(sdk: SDK<API>) {
+    sdk.api.register("multiply", multiply);
+}
+```
+
+::: tip
+To view the entire backend script, expand the following:
+
+<details>
+<summary>Example</summary>
+
+``` ts
+import { SDK, DefineAPI } from "caido:plugin";
+
+function multiply(sdk: SDK, a: number, b: number) {
+    const result = a * b;
+    sdk.console.log(`The product of the multiply call is: ${result}`);
+    return result;
+}
+
+export type API = DefineAPI<{
+    multiply: typeof multiply;
+}>;
+
+export function init(sdk: SDK<API>) {
+    sdk.api.register("multiply", multiply);
+}
+```
+
+</details>
+:::
```

Comment:
I would avoid having code snippets that need to be expanded. If anything, this snippet is the most relevant part for the user, cause they can just copy paste the entire thing and update it as needed.

Can we try having that snippet near the top of the "Registering an API Endpoint" section (no need to wrap it in a `:::tip:::`), then have the explanations for each part underneath?